pkgs/by-name/os/oscar/package.nix

```diff
@@ -32,32 +32,48 @@ stdenv.mkDerivation rec {
 
   qmakeFlags = [ "OSCAR_QT.pro" ];
 
-  installPhase = ''
-    runHook preInstall
-    install -d $out/bin
-    install -d $out/share/OSCAR/Help
-    install -d $out/share/OSCAR/Html
-    install -d $out/share/OSCAR/Translations
-    install -d $out/share/icons/OSCAR
-    install -d $out/share/applications
-    install -T oscar/OSCAR $out/bin/OSCAR
-    # help browser was removed 'temporarily' in https://gitlab.com/pholy/OSCAR-code/-/commit/57c3e4c33ccdd2d0eddedbc24c0e4f2969da3841
-    # install oscar/Help/* $out/share/OSCAR/Help
-    install oscar/Html/* $out/share/OSCAR/Html
-    install oscar/Translations/* $out/share/OSCAR/Translations
-    install -T Building/Linux/OSCAR.png $out/share/icons/OSCAR/OSCAR.png
-    install -T Building/Linux/OSCAR.desktop $out/share/applications/OSCAR.desktop
-    runHook postInstall
-  '';
+  # Graphs/gGraphView.cpp:1416:9: error: variable 'pinned_graphs' set but not used [-Werror,-Wunused-but-set-variable]
+  NIX_CFLAGS_COMPILE = lib.optionals stdenv.cc.isClang [ "-Wno-error=unused-but-set-variable" ];
+
+  installPhase =
+    ''
+      runHook preInstall
+    ''
+    + (
+      if stdenv.isDarwin then
+        ''
+          install -d $out/Applications
+          mv oscar/OSCAR.app $out/Applications
```

Comment:
Also link executable to `$out/bin` and test if it works.

Reply:
From a macOS perspective this seems like an unusual thing to do, and it would make more sense to me to remove the `mainProgram` attribute on Darwin. I haven't really been keeping up with nixpkgs lately, so if that's the convention then I don't mind doing so. The one thing I found when searching ([`mainProgram` for Mac OS Apps](https://discourse.nixos.org/t/mainprogram-for-mac-os-apps/17121)) made it seem like macOS wasn't really considered when implementing the mainProgram feature, and also didn't arrive at a conclusion for what to do in this case.

Reply:
I'm not very sure about that, but I got a request at https://github.com/NixOS/nixpkgs/pull/360354#issuecomment-2510930583. As it doesn't hurt anything and it ease test when reviewing new versions, I'd suggest to add it as well.